source/common/http/filter_manager.cc

```diff
@@ -874,8 +878,17 @@ void DownstreamFilterManager::sendLocalReply(
   if (!filter_manager_callbacks_.responseHeaders().has_value() &&
       !filter_manager_callbacks_.informationalHeaders().has_value()) {
     // If the response has not started at all, send the response through the filter chain.
-    sendLocalReplyViaFilterChain(is_grpc_request, code, body, modify_headers, is_head_request,
-                                 grpc_status, details);
+
+    // We only prepare a local reply to execute later if we're actively
+    // invoking filters to avoid re-entrant in filters.
+    if (avoid_reentrant_filter_invocation_during_local_reply_ &&
+        state_.decoder_filter_chain_aborted_) {
```

Comment:
Is `decoder_filter_chain_aborted_` a way to check whether we are actively invoking filters? Not immediately obvious why we are checking this from the comment